### ISSUE TITLE
Helix having problem with alt+* shortcuts

### ISSUE THREAD

**Issue by last-arg** [cfoust/cy#32]
When using helix noticed that in most cases alt shortcuts were not working. On
some occasions alt shortcuts worked. When alt shortcut didn't work it just used
whatever letter I pressed with alt key.

In terminal command line alt shortcuts work. Like alt+f/b.

Used 'showkey -a' command to see what output 'alt+o' produces.

wezterm (alt + o):
```
^[o      27 0033 0x1b
        111 0157 0x6f
^[o      27 0033 0x1b
        111 0157 0x6f
^[o      27 0033 0x1b
        111 0157 0x6f
```

wezterm with cy (alt + o):
```
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[ o     27 0033 0x1b
        111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[o      27 0033 0x1b
        111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[       27 0033 0x1b
o       111 0157 0x6f
^[ o     27 0033 0x1b
        111 0157 0x6f
```

'ctrl + o' seem to work normally with cy.

wezterm with cy (ctrl + o):
```
^O       15 0017 0x0f
^O       15 0017 0x0f
^O       15 0017 0x0f
```

**Comment by last-arg** [comment]
Yes, linux (nixos)

**Comment by last-arg** [comment]
Did some debugging and I think I see where the problem occurs. At the bottom
of the post there are some strace logs that show what I am talking about.

"alt+o" is split by Input() function in "pkg/bind/module.go". It sends
"alt" and "o" as separate messages. Because they are separate messages they get
written separately. And in most cases read separately. On some lucky cases
they are read together. As can be seen in the strace logs. Because the messages
are read separately they are treated as separate key presses. Maybe on mac
read is done if all/some writes are done? Or something similar.

So need to send alt key combinations as one message. One possible solution that I
can see is to treat alt+<key> as one message that gets written together. Would
need to change DetectOneMsg() in "pkg/taro/key.go" for that. Did not look
deep into if it is possible?

Or some other solution?

-----------------------
strace log: good case
```console
# Read keys pressed - "alt+o"
26905 <... read resumed>"\33o", 32768)  = 2
...
26910 write(12, "\33", 1)               = 1
# Let OS know that want to read stdin
26944 read(0,  <unfinished ...>
# Lucky that "o" is written before stdin reading begins
26910 write(12, "o", 1)                 = 1
26944 <... read resumed>"\33o", 1024)   = 2

```

strace log: bad case
```console
# Read keys pressed - "alt+o"
26905 <... read resumed>"\33o", 32768)  = 2
...
26911 write(12, "\33", 1)               = 1
# Reads from stdin immediately
26944 read(0, "\33", 1024)              = 1
# Now writes "o" 
26911 write(12, "o", 1)                 = 1
...
# Finally "o" gets read
26938 read(0, "o", 1024)                = 1
```

**Comment by cfoust** [comment]
Gosh, I really appreciate you taking a look at this! I think you're right, this is exactly what's going on. The support for alt+ key combinations in cy was an afterthought; I never subjected it to rigorous testing. One of the other contributors and I are going to look into it in a few minutes. I suspect it should be an easy fix.

**Comment by cfoust** [comment]
This is fixed in [v1.3.1](https://github.com/cfoust/cy/releases/tag/v1.3.1)! Thank you for the detailed report.

**Comment by last-arg** [comment]
Awesome. Thanks a lot.

Things seem to work after doing initial test.
